src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.cs

```diff
@@ -54,6 +54,10 @@ public static IEnumerable<object[]> Create_DifferentTfms()
             yield return new object[] { "build_property.TargetFramework = net5.0-ios14.0", true };
             yield return new object[] { "build_property.TargetFramework = Net99", true };
             yield return new object[] { "build_property.TargetFramework = netcoreapp5", false };
+            yield return new object[] { "build_property.TargetFramework = net10.0", true };
+            yield return new object[] { "build_property.TargetFramework = net10", true };
```

Comment:
I'm curious why TargetFrameworkVersion can even not contain something which doesn't parse as `Version` in the first place - we should be consistent what we call Version in all our products, not just .NET, it's weird that we need to do shenanigans with parsing as version, if fails parse as int in order to get this parsed by the framework for that.

Reply:
`MSBuild` itself has to support far more than just .NET and in the rest of product (and general ecosystem) it is completely valid to have a version with `[1, 4]` components. `System.Version` is what is overly restrictive here (and which arguably shouldn't be).

Reply:
Now with that being said, both the `Net99` and `net10` entries here are invalid from the current props perspective for actual inference. I had only added the latter one because we had the former and it ensured the exact bug we're fixing was being handled (which is that it only checked the first version digit).